torcheeg/trainers/classifier.py

```diff
@@ -266,4 +296,349 @@ def predict_step(self,
+            weight_decay (float): The weight decay. (default: :obj:`0.0`)
+            devices (int): The number of devices to use. (default: :obj:`1`)
+            accelerator (str): The accelerator to use. Available options are: 'cpu', 'gpu'. (default: :obj:`"cpu"`)
+            metrics (list of str): The metrics to use. Available options are: 'precision', 'recall', 'f1score', 'accuracy'. (default: :obj:`['accuracy', 'precision', 'recall', 'f1score']`)
+
+        .. automethod:: fit
+        .. automethod:: test
+    '''
+    def __init__(self,
+                 model: nn.Module,
+                 center_dim: int = None,
+                 num_classes: int = None,
+                 lammda: float = 0.0005,
+                 lr: float = 1e-3,
+                 weight_decay: float = 0.0,
+                 devices: int = 1,
+                 accelerator: str = "cpu",
+                 metrics: List[str] = [
+                     'accuracy', 'precision', 'recall', 'f1score'
+                 ]):
+        
+        super(CLossClassifierTrainer,self).__init__(model, num_classes, lr, weight_decay, devices,
+                             accelerator, metrics,center_dim = center_dim) 
+        self.lammda = lammda
+        self.automatic_optimization = False
+        self.decoder,self.Fpredictor= (model.decoder, 
+                                       model.predict_by_feature)
+        self.center_dim = center_dim
+        self.__try_init_center_dim()
+    
+
+        
+
+        
+        
+    def __try_init_center_dim(self):
+        if not self.center_dim :
+            try:
+                self.center_dim = self.model.center_dim
+                if self.num_classes:
+                    self.Closs = CentersLoss(self.num_classes, self.center_dim)
+            except AttributeError:
+                pass
+        else:
+            if self.num_classes:
+                self.Closs = CentersLoss(self.num_classes, self.center_dim)
+
+        
+    
+    def init_metrics(self, metrics: List[str], num_classes: int) -> None:
+        # for train
+        self.train_loss = torchmetrics.MeanMetric()
+        self.center_loss_train = torchmetrics.MeanMetric()
+        self.predict_loss_train = torchmetrics.MeanMetric()
+        
+        # val
+        self.val_loss = torchmetrics.MeanMetric()
+        self.center_loss_val = torchmetrics.MeanMetric()
+        self.predict_loss_val = torchmetrics.MeanMetric()
+
+        #test
+        self.test_loss = torchmetrics.MeanMetric()
+        self.center_loss_test = torchmetrics.MeanMetric()
+        self.predict_loss_test= torchmetrics.MeanMetric()
+
+        # classification metrics for train val test
+        self.train_metrics = classification_metrics(metrics, num_classes)
+        self.val_metrics = classification_metrics(metrics, num_classes)
+        self.test_metrics = classification_metrics(metrics, num_classes)
+    
+    def reset_metric(self, state: str = "train"):
```

Comment:
Avoid over-encapsulation; please place them back in their respective functions. The code snippets aren't being reused; they have simply been amalgamated, which increases the cost of comprehension.